chalice/local.py

```diff
@@ -4,27 +4,99 @@
 
 """
 from __future__ import print_function
+import re
+import time
+import uuid
 import base64
 import functools
+import warnings
 from collections import namedtuple
 
 from six.moves.BaseHTTPServer import HTTPServer
 from six.moves.BaseHTTPServer import BaseHTTPRequestHandler
-from typing import List, Any, Dict, Tuple, Callable  # noqa
+from typing import List, Any, Dict, Tuple, Callable, Optional  # noqa
+
+from chalice.app import Chalice  # noqa
+from chalice.app import CORSConfig  # noqa
+from chalice.app import ChaliceAuthorizer  # noqa
+from chalice.app import RouteEntry  # noqa
+from chalice.app import Request  # noqa
+from chalice.app import AuthResponse  # noqa
+from chalice.app import BuiltinAuthConfig  # noqa
+from chalice.config import Config  # noqa
 
-from chalice.app import Chalice, CORSConfig  # noqa
 from chalice.compat import urlparse, parse_qs
 
 
 MatchResult = namedtuple('MatchResult', ['route', 'captured', 'query_params'])
 EventType = Dict[str, Any]
+ContextType = Dict[str, Any]
+HeaderType = Dict[str, Any]
+ResponseType = Dict[str, Any]
 HandlerCls = Callable[..., 'ChaliceRequestHandler']
 ServerCls = Callable[..., 'HTTPServer']
 
 
-def create_local_server(app_obj, port):
-    # type: (Chalice, int) -> LocalDevServer
-    return LocalDevServer(app_obj, port)
+class Clock(object):
+    def time(self):
+        # type: () -> float
+        return time.time()
+
+
+def create_local_server(app_obj, config, port):
+    # type: (Chalice, Config, int) -> LocalDevServer
+    return LocalDevServer(app_obj, config, port)
+
+
+class LocalArnBuilder(object):
+    ARN_FORMAT = ('arn:aws:execute-api:{region}:{account_id}'
+                  ':{api_id}/{stage}/{method}/{resource_path}')
+    LOCAL_REGION = 'mars-west-1'
```

Comment:
Wanted to make it very clear it wasn't a real ARN and to force it never to match with any real ARNs unless they explicitly don't care about the region (with a glob). I can change it if you have a reason you would prefer a real region.